src/dbnode/storage/shard_race_prop_test.go

```diff
@@ -155,28 +155,35 @@ func TestShardTickWriteRace(t *testing.T) {
 
 	wg.Add(numRoutines)
 
+	doneFn := func() {
+		if r := recover(); r != nil {
+			assert.Fail(t, "unexpected panic: %v", r)
```

Comment:
Maybe better as 
```
assert.NoError(t, recover(), "unexpected panic")
wg.Done()
```
?

Reply:
unfortunately can't be NoError cause the panic doesn't have to be of type `error`. Could make it an `assert.Nil` if you feel strongly